### ISSUE TITLE
Cannot get the Separators to switch to ;

### ISSUE THREAD

**Issue by Kilsgaard** [DaxStudio/DaxStudio#36]
Hi
I have not been able to get the Switch Separators to work when I switch to "OTHER (a;b;c 1,23)".

It is still not possible to run queries with ; instead of ,.

I have a model started in VS 2017, and connects to that in Dax Studio 2.7.1 
I have the following query in my model:
```
DISTINCT (
    SELECTCOLUMNS (
        FILTER (
            DimAccrual;
            DimAccrual[Computation Group] IN { "BANK"; "CERTIFIKAT"; "DIVERSE";"FAST_DIV";"KOSHER";"LEGAL";"TOLD" }
        );
        "Computation Group"; [Computation Group]
    )
)
```

When I puts this query in Dax Studio (have switched to Other in settings), I get the following error:
15-12-2017 09:59:17		Query (5, 23) The syntax for ';' is incorrect. (EVALUATE
```
15-12-2017 10:14:10		Query (5, 23) The syntax for ';' is incorrect. (EVALUATE
DISTINCT (
    SELECTCOLUMNS (
        FILTER (
            DimAccrual;
            DimAccrual[Computation Group] IN { "BANK"; "CERTIFIKAT"; "DIVERSE";"FAST_DIV";"KOSHER";"LEGAL";"TOLD" }
        );
        "Computation Group"; [Computation Group]
    )
)).
```

**Comment by marcosqlbi** [comment]
DAX Studio only supports the comma separator (,) 
This is the same for the DAX engine. The ";" separator is just an artifact of the UI (VS, Power BI), but the underlying model always uses comma.
You can press the button "Swap Delimiters" to converts ";" to "," when you are in DAX Studio, so you can run the query. Press it again, and you revert "," to ";"
I hope it helps!

**Comment by dgosbell** [comment]
Actually, we do support both styles of separators. :) We do this if you've set your options to use the "Other" option by effectively running the code behind the swap separators button before sending the query to the data source (as Tabular data sources only understand the "US" format).  

However, we have synchronous and asynchronous ExecuteQuery methods and somehow the async version (which pretty much everything uses now) is missing this logic. So I can get a fix for this into the next release, I just want to do a bit more refactoring to see if I can consolidate this logic to avoid this sort of issue being repeated in future.

**Comment by marcosqlbi** [comment]
Ops! I didn't realize that DAX Studio support dynamic change of separators! What a shame!!

**Comment by Kilsgaard** [comment]
Hi
Thanks :-)
I am very new to DAX and DAXStudio, had not seen the button to change from ";" to ";" it would be great if it worked against the model 👍 

Please tell me if i can help test when there is an update :-)
and thanks for a great Tool..

**Comment by dgosbell** [comment]
I'm not entirely sure what you mean when you say "worked against the model". Internally the model stores all expressions in the "US" format. It's the Microsoft designer User Interfaces which dynamically switch the delimiters to display using the region of the current workstation.  

If you have example expressions that you want to paste into your model you can first paste them into DAX Studio and switch the delimiters there. If you hoping for a button that does this directly in the designers unfortunately that is not possible at this point in time as Microsoft has not built in support for extensions to the tabular designers.

**Comment by Kilsgaard** [comment]
Sorry for by bad writing.
I did not see the button "Swap Delimiters" that Marco wrote about, at the moment I use this button when I copy DAX formulas from the Tabular model.
But in time it would be nice if it worked as in the Tabular Model.

**Comment by OlaSa2** [comment]
It would be great to support what most countries use ; 
https://en.wikipedia.org/wiki/Decimal_mark#Examples_of_use

**Comment by dgosbell** [comment]
This should be fixed in the 2.7.2 release
